src/async/parallel.ts

```diff
@@ -9,30 +9,46 @@ type WorkItemResult<K> = {
  * Executes many async functions in parallel. Returns the results from
  * all functions as an array. After all functions have resolved, if
  * any errors were thrown, they are rethrown in an instance of
- * AggregateError.
+ * AggregateError. The operation can be aborted by passing optional AbortSignal,
+ * which will throw an Error if aborted.
  *
  * @see https://radashi.js.org/reference/async/parallel
  * @example
  * ```ts
  * // Process images concurrently, resizing each image to a standard size.
+ * const abortController = new AbortController();
  * const images = await parallel(2, imageFiles, async (file) => {
  *   return await resizeImage(file)
- * })
+ * }, abortController)
+ *
+ * // To abort the operation:
+ * // abortController.abort()
  * ```
  * @version 12.1.0
  */
 export async function parallel<T, K>(
   limit: number,
   array: readonly T[],
   func: (item: T) => Promise<K>,
+  signal?: AbortSignal,
```

Comment:
Instead of adding another argument, let's change `limit` to be:
```ts
options: number | ParallelOptions,
```
…with this type:
```ts
export interface ParallelOptions {
  limit: number
  signal?: AbortSignal
}
```
Then add some runtime type-checking:
```ts
if (isNumber(options)) {
  options = { limit: options }
}
```

Reply:
I don't see a commit that addresses this feedback, so this is yet “unresolved”.